Disher/View Controllers/RecipesViewController.m

```diff
@@ -87,38 +115,140 @@ - (void) queryMealDB:(NSString *) name completionHandler:(void(^)(NSArray *retur
 
            }
     }];
     [task resume];
 }
 
+- (void) querySearchSpoonacular:(NSString *) name completionHandler:(void(^)(NSArray *returnedMeals))completionHandler {
+    //Use API Key in Keys.plist file
+    NSString *path = [[NSBundle mainBundle] pathForResource: @"Keys" ofType: @"plist"];
+    NSDictionary *dict = [NSDictionary dictionaryWithContentsOfFile: path];
+    NSString *key = [dict objectForKey: @"spoon_key"];
+    NSString *apiKeyArg = [NSString stringWithFormat:@"&apiKey=%@", key];
+    NSString *queryURL = [NSString stringWithFormat:@"https://api.spoonacular.com/recipes/complexSearch?query=\"%@\"%@", name, apiKeyArg];
+    queryURL = [queryURL stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLQueryAllowedCharacterSet]];
+    
+    NSURL *url = [NSURL URLWithString:queryURL];
+    NSURLRequest *request = [NSURLRequest requestWithURL:url cachePolicy:NSURLRequestReloadIgnoringLocalCacheData timeoutInterval:10.0];
+    NSURLSession *session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration] delegate:nil delegateQueue:[NSOperationQueue mainQueue]];
+    NSURLSessionDataTask *task = [session dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+           if (error != nil) { //TODO: Add error message
+           }
+           else {
+               NSDictionary *dataDictionary = [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingMutableContainers error:nil];
+               if (![(NSArray *)dataDictionary[@"results"] count]) {
+                   NSLog(@"null detected from spoonacular");
+               }
+               else {
+                   completionHandler(dataDictionary[@"results"]);
+               }
+           }
+    }];
+    [task resume];
+}
+
+- (void) queryAPIs:(NSString *) input completionHandler:(void(^)(void))completionHandler {
+    dispatch_group_t group = dispatch_group_create();
+    dispatch_group_enter(group);
+    dispatch_group_async(group, dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^ {
+        [self querySearchSpoonacular:input completionHandler:^(NSArray *returnedMeals) {
+            self.spoonResults = returnedMeals;
+            dispatch_group_leave(group);
+        }];
+    });
+    dispatch_group_enter(group);
+    dispatch_group_async(group, dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^ {
+        [self queryMealDB:input completionHandler:^(NSArray *returnedMeals) {
+            self.mealDBresults = returnedMeals;
+            dispatch_group_leave(group);
+        }];
+    });
+    dispatch_group_notify(group, dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^ {
+        // All group blocks have now completed
+        NSLog(@"completion");
+        dispatch_async(dispatch_get_main_queue(), ^{
+            completionHandler();
+        });
+    });
+}
+
+//Table View Cell Methods
+- (NSArray *)rightButtons
+{
```

Comment:
nit: let's keep the bracket formatting conventions the same as others in the file 😛